### PR TITLE
Convert singlenode-kafka to Launcher

### PR DESCRIPTION
**Pull request by kokosing**
Migrate singlenode-kafka to PTL

### CODE REVIEW

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentModule.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import io.prestosql.tests.product.launcher.env.common.Hadoop;
 import io.prestosql.tests.product.launcher.env.common.Kerberos;
 import io.prestosql.tests.product.launcher.env.common.Standard;
+import io.prestosql.tests.product.launcher.env.environment.SinglenodeKafka;
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentModule.java]: > Migrate singlenode-kafka to PTL

I used this convention for commit messages:

"Convert singlenode-kafka to Launcher"

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/common/Hadoop.java [hunk 2]
```diff
@@ -31,7 +31,9 @@
 public final class Hadoop
         implements EnvironmentExtender
 {
-    public static final String CONTAINER_PRESTO_HIVE_PROPERTIES = "/docker/presto-product-tests/conf/presto/etc/catalog/hive.properties";
+    public static final String CONTAINER_PRESTO_CATALOG = "/docker/presto-product-tests/conf/presto/etc/catalog/";
+    public static final String CONTAINER_PRESTO_HIVE_PROPERTIES = CONTAINER_PRESTO_CATALOG + "hive.properties";
+    public static final String CONTAINER_PRESTO_KAFKA_PROPERTIES = CONTAINER_PRESTO_CATALOG + "kafka.properties";
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/common/Hadoop.java]: This constant doesn't seem to belong here, as `Hadoop` class doesn't have anything to do with `kafka.properties`.

We could define `/docker/presto-product-tests/conf/presto` as a const in `Standard`

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        builder.addContainer("zookeeper", createZookeeper());
+        builder.addContainer("kafka", createKafka());
+    }
+
+    @SuppressWarnings("resource")
+    private DockerContainer createZookeeper()
+    {
+        DockerContainer container = new DockerContainer("confluentinc/cp-zookeeper:" + CONFLUENT_VERSION)
+                .withNetworkAliases("zookeeper")
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java]: redundant

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                .withEnv("ZOOKEEPER_CLIENT_PORT", "2181")
+                .withEnv("ZOOKEEPER_TICK_TIME", "2000")
+                .withExposedPorts(2181);
+        container.setDockerImageName("zookeeper");
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java]: that's unlikely what you want, is it?
(please remove)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    @SuppressWarnings("resource")
+    private DockerContainer createZookeeper()
+    {
+        DockerContainer container = new DockerContainer("confluentinc/cp-zookeeper:" + CONFLUENT_VERSION)
+                .withNetworkAliases("zookeeper")
+                .withEnv("ZOOKEEPER_CLIENT_PORT", "2181")
+                .withEnv("ZOOKEEPER_TICK_TIME", "2000")
+                .withExposedPorts(2181);
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java]: the port gets exposed on localhost at random number

that's unlikely useful. Skip it?
Or do you use it as an implicit health check? (yes =>  add a comment)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    private DockerContainer createKafka()
+    {
+        DockerContainer container = new DockerContainer("confluentinc/cp-kafka:" + CONFLUENT_VERSION)
+                .withNetworkAliases("kafka")
+                .withEnv("KAFKA_BROKER_ID", "1")
+                .withEnv("KAFKA_ZOOKEEPER_CONNECT", "zookeeper:2181")
+                .withEnv("KAFKA_ADVERTISED_LISTENERS", "PLAINTEXT://kafka:9092")
+                .withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", "1")
+                .withEnv("KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS", "0")
+                .withExposedPorts(9092);
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java]: the port gets exposed on localhost at random number

that's unlikely useful. Skip it?
Or do you use it as an implicit health check? (yes =>  add a comment)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        container.setDockerImageName("kafka");
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java]: that's unlikely what you want, is it?
(please remove)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+                .withExposedPorts(2181);
+        container.setDockerImageName("zookeeper");
+        return container;
+    }
+
+    @SuppressWarnings("resource")
+    private DockerContainer createKafka()
+    {
+        DockerContainer container = new DockerContainer("confluentinc/cp-kafka:" + CONFLUENT_VERSION)
+                .withNetworkAliases("kafka")
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java]: redundant

**Comment by kokosing** [comment]
Updated and fixed.

**Pull request closed without merging**